File: client/Client/Network/Messages/Game/PlayerCommandMessage.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace LoadRunnerClient.Messages
{
    /// Author Florian Ortmann
    public class PlayerCommandMessage
    {
        public const string TYPE = "PlayerCommandMessage";
        private PressedKey _pressedKey;

        public PlayerCommandMessage(PressedKey pressedKey)
        {
            this._pressedKey = pressedKey;
        }

        [JsonConverter(typeof(StringEnumConverter))]
        public PressedKey pressedKey
        {
            get => _pressedKey;
        }
    }

    public enum PressedKey
    {
        LEFT,
        RIGHT,
        UP,
        DOWN,
        JUMP,
        DIG_LEFT,
        DIG_RIGHT,
        ENERGY
    }
}

File: server/src/main/java/de/hsrm/mi/swtpro03/server/network/messages/game/PlayerCommandMessage.java
package de.hsrm.mi.swtpro03.server.network.messages.game;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;

public class PlayerCommandMessage {

	public static final String TYPE = "PlayerCommandMessage";

	private PressedKey pressedKey;

	public PlayerCommandMessage() {
	}

	public PlayerCommandMessage(@JsonProperty("pressedKey") String pressedKey) {
		this.pressedKey = PressedKey.valueOf(pressedKey);
	}

	public PressedKey getPressedKey() {
		return pressedKey;
	}

	@JsonFormat(shape = JsonFormat.Shape.OBJECT)
	public enum PressedKey {
		LEFT,
		RIGHT,
		UP,
		DOWN,
		JUMP,
		DIG_LEFT,
		DIG_RIGHT,
		ENERGY
	}
}
